Framework/MDAlgorithms/test/IntegrateEllipsoidsTwoStepTest.h

```diff
@@ -0,0 +1,410 @@
+#include "MantidMDAlgorithms/IntegrateEllipsoidsTwoStep.h"
+
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/DetectorInfo.h"
+#include "MantidKernel/NearestNeighbours.h"
+#include "MantidDataObjects/PeaksWorkspace.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidGeometry/Crystal/OrientedLattice.h"
+#include "MantidKernel/V3D.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+#include "MantidTestHelpers/SingleCrystalDiffractionTestHelper.h"
+
+#include <cxxtest/TestSuite.h>
+#include <tuple>
+#include <random>
+
+using namespace Mantid;
+using namespace Mantid::MDAlgorithms;
+using namespace Mantid::DataObjects;
+using Mantid::Kernel::V3D;
+using Mantid::Geometry::OrientedLattice;
+using namespace Mantid::SingleCrystalDiffractionTestHelper;
+
+class IntegrateEllipsoidsTwoStepTest : public CxxTest::TestSuite {
+
+public:
+  void test_init() {
+    Mantid::MDAlgorithms::IntegrateEllipsoidsTwoStep alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+  }
+
+  void test_exec_events_with_no_background() {
+    const int numEventsPerPeak = 10000;
+    // Very tight distribution with events happening at a single point
+    const auto sigmas = std::make_tuple(.002, .002, 0.1);
+
+    // Build some diffraction data
+    WorkspaceBuilder builder;
```

Comment:
Could do. But I'd argue that having the setup for each test close to the test and assertions makes it clearer what the current test environment is. Making it a function would make it more brittle and harder to customise for individual tests. For example I think some peaks have lower counts in tests further down.